lambdas/check-coi/src/main/java/uk/gov/di/ipv/core/checkcoi/CheckCoiHandler.java

```diff
@@ -0,0 +1,137 @@
+            VerifiableCredentialService verifiableCredentialService,
+            SessionCredentialsService sessionCredentialsService,
+            UserIdentityService userIdentityService) {
+        this.configService = configService;
+        this.ipvSessionService = ipvSessionService;
+        this.clientOAuthSessionDetailsService = clientOAuthSessionDetailsService;
+        this.verifiableCredentialService = verifiableCredentialService;
+        this.sessionCredentialsService = sessionCredentialsService;
+        this.userIdentityService = userIdentityService;
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public CheckCoiHandler() {
+        this.configService = new ConfigService();
+        this.ipvSessionService = new IpvSessionService(configService);
+        this.clientOAuthSessionDetailsService = new ClientOAuthSessionDetailsService(configService);
+        this.verifiableCredentialService = new VerifiableCredentialService(configService);
+        this.sessionCredentialsService = new SessionCredentialsService(configService);
+        this.userIdentityService = new UserIdentityService(configService);
+    }
+
+    @Override
+    @Tracing
+    @Logging(clearState = true)
+    public Map<String, Object> handleRequest(ProcessRequest request, Context context) {
+        configService.setFeatureSet(RequestHelper.getFeatureSet(request));
+        LogHelper.attachComponentId(configService);
+
+        try {
+            var ipvSession =
+                    ipvSessionService.getIpvSession(RequestHelper.getIpvSessionId(request));
+            var clientOAuthSession =
+                    clientOAuthSessionDetailsService.getClientOAuthSession(
+                            ipvSession.getClientOAuthSessionId());
+            LogHelper.attachGovukSigninJourneyIdToLogs(
+                    clientOAuthSession.getGovukSigninJourneyId());
+
+            var userId = clientOAuthSession.getUserId();
+
+            var credentials =
+                    Stream.concat(
+                                    verifiableCredentialService.getVcs(userId).stream(),
+                                    sessionCredentialsService
+                                            .getCredentials(ipvSession.getIpvSessionId(), userId)
+                                            .stream())
+                            .toList();
+
+            var coiSubjourneyType = ipvSession.getCoiSubjourneyType();
+            var successfulCheck =
+                    switch (coiSubjourneyType) {
+                        case GIVEN_NAMES_ONLY, GIVEN_NAMES_AND_ADDRESS -> userIdentityService
+                                .areFamilyNameAndDobCorrelatedForCoiCheck(credentials);
+                        case FAMILY_NAME_ONLY, FAMILY_NAME_AND_ADDRESS -> userIdentityService
+                                .areGivenNamesAndDobCorrelated(credentials);
+                        case ADDRESS_ONLY -> userIdentityService.areVcsCorrelated(credentials);
```

Comment:
Might be good practice to put a default case in (false) ?

Reply:
I'm not 100% sure it's necessary - it currently covers all enum's, and if we extend the enum the switch will fail to compile unless we handle it.
It might make it more obvious that we're covering all bases though.